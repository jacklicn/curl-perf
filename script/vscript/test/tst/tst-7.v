
sub fact(a)

  if a == 1 
    return 1
  end

  return a * fact(a-1)
end

sub fact_loop(a)
  var res = 1
  
  while a > 0
    res = res * a
    a = a - 1
  end
  return res
end

var v = fact(10)
print( v )

var v1 = fact_loop(10)
print( v1 )

//v = fact( fact(4) )
//print( v )

//print( fact(10) )


