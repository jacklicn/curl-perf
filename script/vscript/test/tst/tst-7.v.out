*** syntax ok ***
*** semantics and code generation ok ***

>3628800

>3628800
